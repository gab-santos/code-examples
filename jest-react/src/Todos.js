import React, { Component } from "react";

import { connect } from "react-redux";

const mapStateToProps = state => ({
  todos: state.todos
});

class Todos extends Component {
  render() {
    return (
      <div className="Todos">
        {this.props.todos.map(todo => (
          <li key={todo.id}>{todo.text}</li>
        ))}
      </div>
    );
  }
}

export default connect(mapStateToProps)(Todos);
