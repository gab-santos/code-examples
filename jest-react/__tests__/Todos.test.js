// Testar componentes quando eles se modificam baseados em alguma circunstância

import React from "react";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import Todos from "../src/Todos";

import configureStore from "redux-mock-store";
const mockStore = configureStore();

const initialState = {
  todos: [
    { id: 0, text: "Fazer café" },
    { id: 1, text: "Acessar o slack" },
    { id: 2, text: "Ver RocketLive" }
  ]
};

const store = mockStore(initialState);

describe("Testing Todos Component", () => {
  it("should render correctly", () => {
    const wrapper = mount(
      <Provider store={store}>
        <Todos />
      </Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
