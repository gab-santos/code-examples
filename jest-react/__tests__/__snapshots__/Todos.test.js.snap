// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Todos Component should render correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Todos)>
    <Todos
      dispatch={[Function]}
      todos={
        Array [
          Object {
            "id": 0,
            "text": "Fazer café",
          },
          Object {
            "id": 1,
            "text": "Acessar o slack",
          },
          Object {
            "id": 2,
            "text": "Ver RocketLive",
          },
        ]
      }
    >
      <div
        className="Todos"
      >
        <li
          key="0"
        >
          Fazer café
        </li>
        <li
          key="1"
        >
          Acessar o slack
        </li>
        <li
          key="2"
        >
          Ver RocketLive
        </li>
      </div>
    </Todos>
  </Connect(Todos)>
</Provider>
`;
